DDD-11.md

```diff
@@ -0,0 +1,165 @@
+
+```java
+final ChronicleQueue queue = SingleChronicleQueueBuilder.single(Testing.Files.datadir()).build();
+try (ExcerptAppender appender = queue.createAppender()) {
+    try (DocumentContext context = appender.writeDocument()) {
+        context.wire()
+            .write("xid").writeString(transactionId)
+            .write("type").writeInt(eventType)
+            ...;
+    }
+} 
+```
+
+Conceptually, each event would have a defined structure of fields and values that would be appended.
+The queue API supports writing boolean, byte, char, short, int, long, float, double, or string data types.
+In addition, you can also write a series of bytes using compression for smaller storage requirements, which would be great for things like clob, blob, and perhaps string data types.
+
+In pseudo code, the transaction reader thread might look something like this:
+
+```java
+executor.submit(() -> {
+    while (context.isRunning()) {
+        final ResultSet changes = fetchChangesFromSourceDatabase();
+        while (changes.next()) {
+            try (DocumentContext dc = queue.getDocumentContext()) {
+                WireOut wire = dc.wire()
+                        .write("xid").writeString(changes.getString(TRANSACTION_ID))
+                        .wire("type").writeInt(changes.getInt(EVENT_TYPE));
+                
+                switch (EventType.from(changes.getInt(EVENT_TYPE))) {
+                    INSERT, UPDATE, DELETE -> wire("sql").writeString(changes.getString(REDO_SQL));
+                    SELECT_LOB_LOCATOR -> ...;
+                    LOB_WRITE -> ...;
+                    XML_BEGIN -> ...;
+                    ...
+                }   
+            }
+        }
+    }
+});
+```
+
+Note: The above code shows the Chronicle API to illustrate its simplicity. Ideally we would prefer to place this behind some abstraction.
+
+The Chronicle Queue API also supports writing objects that extend `SelfDescribingMarshallable`, allowing for a more object-oriented driven approach.
+In this way, you would choose to deserialize each JDBC row into an object representation and then write that to the queue instead.
+
+When it comes to read position offset management, there are two ways the reader thread could manage this:
+
+* Position state is transient. Upon connector restart, the queue files are removed from disk, and reading resumed from the connector offset position (like today).
+* Position state is persistent, and upon restart, the resume position is based on the last read position the thread captured. If we maintain a second offset value for the reader position, we could choose to fallback to the offset position like we use today if there are no queue files persisted on disk, or resume from the last read position if queue files do. This provides the most efficient management and integration between relying on the persisted queues versus re-reading data that may no longer be available.
```

Comment:
I'd aim for two stage implementation. The first solution is the most straightforward and will be great in validation the concept. The additional state management should be added only later.